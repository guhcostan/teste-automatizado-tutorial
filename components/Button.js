import PropTypes from 'prop-types'
import React from 'react'

const Button = props => (
  <button onClick={props.disabled ? props.onClick : null}>
    {props.text}
  </button>
)

Button.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  text: PropTypes.string,
}

export default Button