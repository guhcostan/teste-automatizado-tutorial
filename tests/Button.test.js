import React from 'react'
import toJson from 'enzyme-to-json'
import Button from '../components/Button'
import { mount } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

it('should render', () => {
    const spy = jest.spyOn(global.console, 'error')
    const onClick = jest.fn()
    let props = {
        text: 'Teste',
        onClick: onClick,
        disabled: false
    }
    const app = mount( 
        <Button {...props} />,
    )
    expect(toJson(app)).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()
})

it('should click button', () => {
    const spy = jest.spyOn(global.console, 'error')
    const onClick = jest.fn()
    let props = {
        text: 'Teste',
        onClick: onClick,
        disabled: false
    }
    const app = mount( 
        <Button {...props} />,
    )
    app.simulate('click')
    expect(onClick).toHaveBeenCalled()
    expect(spy).not.toHaveBeenCalled()
})

it('should click button disabled', () => {
    const spy = jest.spyOn(global.console, 'error')
    const onClick = jest.fn()
    let props = {
        text: 'Teste',
        onClick: onClick,
        disabled: true
    }
    const app = mount( 
        <Button {...props} />,
    )
    app.simulate('click')
    expect(onClick).not.toHaveBeenCalled()
    expect(spy).not.toHaveBeenCalled()
})