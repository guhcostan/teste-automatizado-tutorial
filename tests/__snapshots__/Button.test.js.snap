// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<Button
  disabled={false}
  onClick={[MockFunction]}
  text="Teste"
>
  <button
    onClick={[MockFunction]}
  >
    Teste
  </button>
</Button>
`;
